oauth/api/router.py

```diff
@@ -0,0 +1,167 @@
+from logging import getLogger
+from typing import List
+
+from fastapi import APIRouter, Body, Depends, Request, Security, status
+from fastapi.security import HTTPBasic
+from sqlalchemy.orm import Session
+
+import oauth.api.endpoints as endpoints
+from oauth.api.exceptions import (
+    AuthenticationException,
+    ClientNotFoundException,
+    InvalidScopeException,
+)
+from oauth.api.models import (
+    AccessToken,
+    ClientCreatedResponse,
+    OAuth2ClientCredentialsRequestForm,
+)
+from oauth.api.utils import verify_oauth_client
+from oauth.database.models import ClientDetail
+from oauth.scopes import (
+    CLIENT_CREATE,
+    CLIENT_DELETE,
+    CLIENT_READ,
+    CLIENT_UPDATE,
+    SCOPE_READ,
+    SCOPES,
+)
+
+logger = getLogger(__name__)
+router = APIRouter(prefix=endpoints.OAUTH_PREFIX, tags=["OAuth"])
+
+
+@router.post(endpoints.TOKEN, response_model=AccessToken)
+async def acquire_access_token(
+    request: Request,
+    form_data: OAuth2ClientCredentialsRequestForm = Depends(),
+    db: Session = Depends(),
```

Comment:
We'll need to find a way to replicate the behavior of [fidesops' `get_db` helper](https://github.com/ethyca/fidesops/blob/f1e0472eef4bd3287486b56c175a8343f487c1ab/src/fidesops/api/deps.py#L7) here, and anywhere else there's a `Session` dependency callback.

Reply:
As an aside, I don't love how we have to pass the db into every function where we need it in fidesops.  The [`get_db` helper is a FastAPI recommendation](https://fastapi.tiangolo.com/tutorial/sql-databases/#create-a-dependency ) that gets a session per request and closes it after it's completed. 

[With Flask, alternatively](https://flask.palletsprojects.com/en/2.0.x/tutorial/database/#connect-to-the-database)    you don't have to pass the session to each function, Flask has a `g` object per request where you can store and reuse the session if it's called more than once in a request.   You can just import the `get_db` method in the file which gets the session off of g if it exists and creates if it it doesn't.  

https://github.com/tiangolo/fastapi/issues/2894#issuecomment-791262048 was using contextvars to do something similar in FastAPI. I'm not saying we need to change this now, it's not the easiest thing, just wanted to bring it up as we're propagating decisions to other repos.